frontend/packages/ceph-storage-plugin/src/components/modals/volume-snapshot-modal/volume-snapshot-modal.tsx

```diff
@@ -29,35 +38,201 @@ export type VolumeSnapshotModalProps = {
+        onChange={setScheduleLabel}
+        className="ceph-volume-snapshot-modal--label"
+        placeholder="* * 5 * *"
+      />
+    </FormGroup>
+  );
+
+  const OtherElements: React.FC = () => (
+    <div className="ceph-volume-snapshot-modal--inline">
+      {scheduleType === snapshotTypes.Monthly ? <MonthlyElements /> : <WeeklyElements />}
+      <FormGroup
+        className="ceph-volume-snapshot-modal--input"
+        fieldId="snapshot-modal__time-week"
+        label="Time"
+      >
+        <TextInput
+          type="time"
+          name="snapshot-modal__time-week"
+          aria-label="snapshot-modal-time-week"
+          value={snapshotTime}
+          onChange={setSnapshotTime}
+          className="ceph-volume-snapshot-modal--label"
+        />
+      </FormGroup>
+    </div>
+  );
+
+  const ScheduleElements: React.FC = () => (
+    <Form>
+      {scheduleType === snapshotTypes.CronJob ? <CronElements /> : <OtherElements />}
+      <FormGroup
+        className="ceph-volume-snapshot-modal--input__group"
+        label="Keep"
+        fieldId="snapshot-modal__count"
+        helperText="*Older snapshots will be deleted automatically"
+        isRequired
+      >
+        <div className="ceph-volume-snapshot-modal--input__keep">
+          <Dropdown
+            items={numberOfSnapshot}
+            selectedKey={snapshotCount}
+            onChange={(value) => setSnapshotCount(Number(value))}
+            className="ceph-volume-snapshot-modal--count"
+          />
+          <span className="ceph-volume-snapshot-modal--span_keep">last snapshots</span>
+        </div>
+      </FormGroup>
+    </Form>
+  );
+
   const submit = (event: React.FormEvent<EventTarget>) => {
     event.preventDefault();
     const ns = getNamespace(resource);
     const pvcName = getName(resource);
-    const snapshotTemplate: K8sResourceKind = {
-      apiVersion: VolumeSnapshotModel.apiVersion,
-      kind: VolumeSnapshotModel.kind,
-      metadata: {
-        name: snapshotName,
-        namespace: ns,
-      },
-      spec: {
-        source: {
-          persistentVolumeClaimName: pvcName,
+    if (scheduleType === snapshotTypes.Single) {
+      const snapshotTemplate: K8sResourceKind = {
+        apiVersion: VolumeSnapshotModel.apiVersion,
+        kind: VolumeSnapshotModel.kind,
+        metadata: {
+          name: snapshotName,
+          namespace: ns,
         },
-      },
-    };
-    handlePromise(k8sCreate(VolumeSnapshotModel, snapshotTemplate))
-      .then(close)
-      .catch((error) => {
-        throw error;
-      });
+        spec: {
+          source: {
+            persistentVolumeClaimName: pvcName,
+          },
+        },
+      };
+      handlePromise(k8sCreate(VolumeSnapshotModel, snapshotTemplate))
+        .then(close)
+        .catch((error) => {
+          // eslint-disable-next-line no-console
+          console.error(error);
+        });
+    } else {
+      const pvcLabel = Math.random()
+        .toString(36)
```

Comment:
You should add a comment, why 36 is used here. Can be addressed in a follow up PR.